File: HRMS/HRMS.Web/Modules/ServerTypes/Communication/SocialPostColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SocialPlatform } from "./SocialPlatform";
import { SocialPostRow } from "./SocialPostRow";

export interface SocialPostColumns {
    SocialPostId: Column<SocialPostRow>;
    SocialAccountName: Column<SocialPostRow>;
    SocialAccountPlatform: Column<SocialPostRow>;
    Title: Column<SocialPostRow>;
    PostType: Column<SocialPostRow>;
    Status: Column<SocialPostRow>;
    ScheduledDate: Column<SocialPostRow>;
    PublishedDate: Column<SocialPostRow>;
}

export class SocialPostColumns extends ColumnsBase<SocialPostRow> {
    static readonly columnsKey = 'Communication.SocialPost';
    static readonly Fields = fieldsProxy<SocialPostColumns>();
}

[SocialPlatform]; // referenced types

File: HRMS/HRMS.Web/Modules/Communication/SocialPost/SocialPostColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Communication.Columns;

[ColumnsScript("Communication.SocialPost")]
[BasedOnRow(typeof(SocialPostRow), CheckNames = true)]
public class SocialPostColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int SocialPostId { get; set; }
    public string SocialAccountName { get; set; }
    public SocialPlatform SocialAccountPlatform { get; set; }
    [EditLink]
    public string Title { get; set; }
    public SocialPostType PostType { get; set; }
    public SocialPostStatus Status { get; set; }
    public DateTime ScheduledDate { get; set; }
    public DateTime PublishedDate { get; set; }
}
